.changelog/unreleased/breaking-changes/960-simplify-length-validation.md

```diff
@@ -0,0 +1,3 @@
+- Remove IdentifierError::Empty variant.  It was a special case of
```

Comment:
Please open an issue for this PR and update it here. Your PR description looks fine, so you can just copy-paste it.

Reply:
Once you created your issue, you can leave the PR description as `closes #<issue-no>`.

Reply:
Yes, I’ve already done that.